### PR TITLE
Refactored cmdline common

### PR DESCRIPTION
**Pull request by akallabeth**
* Add settings setter where possible
* Load dynamic sound channel alongside static one
* Load clipboard channel if requested

### CODE REVIEW

File: client/common/cmdline.c [hunk 1]
```diff
@@ -3857,6 +3863,14 @@ BOOL freerdp_client_load_addins(rdpChannels* channels, rdpSettings* settings)
 		}
 	}
 
+	if (settings->RedirectClipboard)
+	{
```
Comment (hardening) [inline client/common/cmdline.c:3867]: Isn't the clipboard channel loaded by the code below when the `RedirectClipboard` setting is set ?

Comment (akallabeth) [reply client/common/cmdline.c]: @hardening true.

**Review by hardening — APPROVED** [review]
Apart from my suggestion, LGTM

File: client/common/cmdline.c [hunk 2]
```diff
@@ -1119,89 +1119,90 @@ BOOL freerdp_parse_hostname(const char* hostname, char** host, int* port)
...
-		settings->DisableFullWindowDrag = FALSE;
-		settings->DisableMenuAnims = FALSE;
-		settings->DisableThemes = FALSE;
-		settings->NetworkAutoDetect = TRUE;
-
-		/* Automatically activate GFX and RFX codec support */
+	if (!freerdp_settings_set_uint32(settings, FreeRDP_ConnectionType, type))
+		return FALSE;
+
+	switch (type)
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply client/common/cmdline.c:1167]: ok, done.

**Pull request merged**